external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutConfig.java

```diff
@@ -60,123 +68,197 @@
+            this.pollTimeoutMs = builder.pollTimeoutMs;
+            this.offsetCommitPeriodMs = builder.offsetCommitPeriodMs;
+            this.maxRetries = builder.maxRetries;
+            this.firstPollOffsetStrategy = builder.firstPollOffsetStrategy;
+            this.maxUncommittedOffsets = builder.maxUncommittedOffsets;
+            //this could result in a lot of class case exceptions at runtime,
+            // but this is the only way to really maintain API compatibility
+            this.translator = (RecordTranslator<K, V>) builder.translator;
+            this.retryService = builder.retryService;
+            this.keyDes = keyDes;
+            this.valueDes = valueDes;
         }
 
         /**
          * Specifying this key deserializer overrides the property key.deserializer
          */
-        public Builder<K,V> setKeyDeserializer(Deserializer<K> keyDeserializer) {
-            this.keyDeserializer = keyDeserializer;
-            return this;
+        public <NK> Builder<NK,V> setKeyDeserializer(Deserializer<NK> keyDeserializer) {
+            return new Builder<>(this, keyDeserializer, valueDes);
         }
 
         /**
          * Specifying this value deserializer overrides the property value.deserializer
          */
-        public Builder<K,V> setValueDeserializer(Deserializer<V> valueDeserializer) {
-            this.valueDeserializer = valueDeserializer;
+        public <NV> Builder<K,NV> setValueDeserializer(Deserializer<NV> valueDeserializer) {
+            return new Builder<>(this, this.keyDes, valueDeserializer);
+        }
+        
+        /**
+         * Set a Kafka property config
+         */
+        public Builder<K,V> setProp(String key, Object value) {
+            kafkaProps.put(key, value);
             return this;
         }
+        
+        /**
+         * Set multiple Kafka property configs
+         */
+        public Builder<K,V> setProp(Map<String, Object> props) {
+            kafkaProps.putAll(props);
+            return this;
+        }
+        
+        /**
+         * Set multiple Kafka property configs
+         */
+        public Builder<K,V> setProp(Properties props) {
+            for (String name: props.stringPropertyNames()) {
+                kafkaProps.put(name, props.get(name));
+            }
+            return this;
+        }
+        
+        /**
+         * Set the group.id for the consumers
+         */
+        public Builder<K,V> setGroupId(String id) {
+            return setProp("group.id", id);
+        }
+        
+        /**
+         * reset the bootstrap servers for the Consumer
+         */
+        public Builder<K,V> setBootstrapServers(String servers) {
+            return setProp(BOOTSTRAP_SERVERS_CONF, servers);
+        }
+        
+        /**
+         * The minimum amount of data the broker should return for a fetch request.
+         */
+        public Builder<K,V> setFetchMinBytes(int bytes) {
+            return setProp("fetch.min.bytes", bytes);
```

Comment:
Nit: Consider using some of the constants in `org.apache.kafka.clients.consumer.ConsumerConfig` instead